--- jsoncpp/jsoncpp-src-0.6.0-rc2/SConstruct	2010-03-13 12:59:50.000000000 +0200
+++ jsoncpp-src-0.6.0-rc2/SConstruct	2013-05-26 19:14:31.257788348 +0300
@@ -117,6 +117,17 @@
 elif platform == 'mingw':
     env.Tool( 'mingw' )
     env.Append( CPPDEFINES=[ "WIN32", "NDEBUG", "_MT" ] )
+elif platform == 'mingw-cross':
+    env.Tool( 'mingw' )
+    mingw_cross_prefix = ARGUMENTS.get('MINGW_CROSS_PREFIX', '')
+    if mingw_cross_prefix: 
+        env['CXX'] = mingw_cross_prefix + "g++"
+        env['CC'] = mingw_cross_prefix + "g++"
+        env['LD'] = mingw_cross_prefix + "ld"
+        env['RANLIB'] = mingw_cross_prefix + "ranlib"
+        env['AR'] = mingw_cross_prefix + "ar"
+    env.Append( CPPDEFINES=[ "WIN32", "NDEBUG", "_MT" ] )
+    env['SHARED_LIB_ENABLED'] = True
 elif platform.startswith('linux-gcc'):
     env.Tool( 'default' )
     env.Append( LIBS = ['pthread'], CCFLAGS = "-Wall" )
